cdap-data-fabric/src/main/java/co/cask/cdap/spi/data/Row.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright © 2018 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.spi.data;
+
+import javax.annotation.Nullable;
+
+/**
+ * Represents a row of data from a {@link StructuredTable}.
+ */
+public interface Row {
+  /**
+   * @return the value of the field named fieldName as integer.
+   * @throws InvalidFieldException if the fieldName is not part of the table schema, or the type does not match.
+   */
+  @Nullable
+  Integer getInteger(String fieldName) throws InvalidFieldException;
+
+  /**
+   * @return the value of the field named fieldName as long.
+   * @throws InvalidFieldException if the fieldName is not part of the table schema, or the type does not match.
+   */
+  @Nullable
+  Long getLong(String fieldName) throws InvalidFieldException;
+
+  /**
+   * @return the value of the field named fieldName as string.
+   * @throws InvalidFieldException if the fieldName is not part of the table schema, or the type does not match.
```

Comment:
Will there be any stringify conversion for numeric type?

Reply:
No stringify conversions. I clarified it in the javadoc